starter_code/src/movies.js

```diff
@@ -2,14 +2,399 @@
+    duration: "2h 34min",
+    genre: ["Crime", "Drama"],
+    rate: 8.9
+  },
+  {
+    title: "The Lord of the Rings: The Return of the King",
+    year: 2003,
+    director: "Peter Jackson",
+    duration: "3h 21min",
+    genre: ["Adventure", "Drama", "Fantasy"],
+    rate: 8.9
+  },
+  {
+    title: "Il buono, il brutto, il cattivo",
+    year: 1966,
+    director: "Sergio Leone",
+    duration: "3h 2min",
+    genre: ["Western"],
+    rate: 8.9
+  }
+];
+let sortedMoviesByYear = orderByYear(unsortedMovies);
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+function howManyMovies(movies) {
+  let movieCount = movies.reduce(function(accumulator, movie) {
+    if (
+      movie.director === "Steven Spielberg" &&
+      movie.genre.includes("Drama")
+    ) {
+      return accumulator + 1;
+    }
+    return accumulator;
+  }, 0);
+  return movieCount;
+}
+
+let count = howManyMovies(sortedMoviesByYear);
+console.log(count);
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically(moviesToOrder) {
+  orderedMovies = [];
+  for (movie of moviesToOrder) {
+    orderedMovies.push(movie.title);
+  }
+  orderedMovies.sort(function(a, b) {
+    return a.localeCompare(b);
+  });
+  if (moviesToOrder.length >= 20) {
+    let firstTwentyMovies = orderedMovies.slice(0, 20);
+    return firstTwentyMovies;
+  }
+  return orderedMovies;
+}
+
+let sortedMoviesByTitle = orderAlphabetically(unsortedMovies);
+console.log(sortedMoviesByTitle);
+
 // Iteration 4: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(movies) {
+  if (movies.length === 0) {
+    return 0;
+  }
+  //   let movieCount = 0;
+  let averageRating =
+    movies.reduce(function(accumulator, movie) {
+      if (movie.rate === undefined) {
+        return accumulator;
+      }
+      // movieCount ++;
+      return accumulator + movie.rate;
+    }, 0) / movies.length; // this doesn't make sense... because if a movie doesn't have a rating it shouldn't affect the average rating. so the average should ignore those movies alltogether and movies.length should be reduced. But Jasmine wasn't happy with that ¯\_(ツ)_/¯
+  return parseFloat(averageRating.toFixed(2));
+}
+
+console.log(ratesAverage(unsortedMovies));
+
 // Iteration 5: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(movies) {
+  console.log(movies);
+  let dramaMovies = movies.filter(function(movie) {
+    // console.log(movie.genre);
+    // console.log(movie.genre.includes("Drama"))
+    return movie.genre.includes("Drama");
+  });
+  return ratesAverage(dramaMovies);
+}
+
 // Iteration 6: Time Format - Turn duration of the movies from hours to minutes
 
+function turnHoursToMinutes(movies) {
+  let moviesInMinutes = [...movies];
+
+  let movieDurations = moviesInMinutes.map(function(movie) {
+    let timeValues = movie.duration.split(" ");
```

Comment:
To get rid of the .split is not a function error you can write this line above line 336:
let movieDur =  movie.duration + ''
let timeValues = movieDur.spli(" ")

The reason for this is probably because movie.duration is a number and by adding '' to it, it'll be converted to a string. I belive .toString() will also work. This also solves the "must return a number" fault in Jasmine